script.js

```diff
@@ -0,0 +1,164 @@
+    function signupForm(){
+
+        document.getElementById("signup").style.display="block";
+        document.getElementById("login").style.display="none";
+    }
+    function signinForm(){
+        document.getElementById("signup").style.display="none";
+        document.getElementById("login").style.display="block";
+    }
+    function validateForm(event){
+        event.preventDefault();
+        const username=document.getElementById("username").value;
+        const email=document.getElementById("email").value;
+        const phoneNumber=document.getElementById("phonenumber").value;
+        const password=document.getElementById("password").value;
+        let encriptedPassword = crypt.encrypt(password);
+        if (!usernameValid(username)){
+            document.getElementById("username").style.borderColor="red";
+            alert("Name should contain atleast 3 letters and numbers are not allowed");
+            document.getElementById("username").style.borderColor="white";
+            return false;
+        }
+        if (!emailValid(email) || !emailNotExist(email)){
+            document.getElementById("email").style.borderColor="red";
+            alert("invalid email");
+            document.getElementById("email").style.borderColor="white";
+            return false;
+        }
+        if (!passwordValid(password)){
+            document.getElementById("password").style.borderColor="red";
+            alert("password should contain atlest 8characters including one uppercase,number and special character");
+            document.getElementById("password").style.borderColor="white";
+            return false;
+        }
+        if(!phoneNumberValid(phoneNumber) ||!phonenumNotExist(phoneNumber) ){
+            document.getElementById("phonenumber").style.borderColor="red";
+            alert("phone number should contain 10 numbers");
+            document.getElementById("phonenumber").style.borderColor="white";
+            return false;
+        }
+        addDeatails(username,email,phoneNumber,encriptedPassword);
+        document.getElementById("signup").reset();
+        return true;
+    }
+
+    function validateDetails(event){
+        event.preventDefault();
+        const existingusername=document.getElementById("existingusername").value;
+        const existingPassword=document.getElementById("existingPassword").value;
+        const userExist=userDetails.filter(obj=>obj.userEmail===existingusername);
+        const decriptPassword=crypt.decrypt(userExist[0].userPassword);
+        if (decriptPassword === existingPassword){
+            const user=userExist[0].userName;
+            const idString=userExist[0].id;
+            const userNameid="userName"+idString;
+            setCookie(userNameid,user,1);
+            document.getElementById("login").reset();
+            window.location.href = "https://aiswarya-a-m.github.io/login/dashboard.html";   
```

Comment:
do path-level routing instead of a full redirect. this won't work if this code is hosted under a different domain.
```suggestion
            window.location.href = "/login/dashboard.html";   
```